lib/handlers/footnote-reference.js

```diff
@@ -9,7 +9,7 @@ function footnoteReference(h, node) {
 
   return h(node.position, 'sup', {id: 'fnref-' + identifier}, [
     h(node, 'a', {href: '#fn-' + identifier, className: ['footnote-ref']}, [
-      u('text', identifier)
+      u('text', node.label)
```

Comment:
```suggestion
      u('text', node.label || identifier)
```

`label` is a bit new, and this project existed before it was introduced. However, the spec states that the [`label` field is optional](https://github.com/syntax-tree/mdast#association), and I think it makes sense to use default to identifier (also for backcompat).